### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by algonell**
Fix typos in documentation, comments, etc.

### CODE REVIEW

**Comment by picnixz** [comment]
For the `configure` script and auto-generated Makefiles, you should instead use `make regen-configure` and `make regen-all` to be sure that all files are up-to-date.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -84,8 +84,7 @@ jobs:
         run: autoreconf -ivf -Werror
       - name: Build CPython
         run: |
-          make -j4 regen-all
-          make regen-stdlib-module-names regen-sbom
+          make -j4 regen-configure regen-all
```
Comment (hugovk) [inline .github/workflows/build.yml]: Why change this? Please revert.

**Comment by picnixz** [comment]
@algonell Maybe I wasn't clear enough but by `make regen-all`, I meant that you needed to run them locally. This will produce new files (namely, a new `configure` script will be generated automatically). 

Note that you either need podman or docker to regenerate the configure script since it depends on a speciifc version of autoconf. If you have trouble with this, I can make it myself and open a pull request on your branch so that you can update it (but everytime you make a change to `configure.ac`, I'll need to re-generate the files).

On platforms with shell scripts (Linux/MacOS), you can do the following:

```sh
./Tools/build/regen-configure.sh     # create an up-to-date 'configure'
./configure                          # create an up-to-date 'Makefile'
make -j12 regen-all
``` 

Then, you can verify the result via `git status`. If everything worked well, you should have some new files to commit. Once you've committed them and pushed them, the CI should be green.

---

Slightly off-topic, but you don't need to merge `main` into your branch unless you have conflicts. This will notify people watching the PR (like me) and we think that there is a relevant commit to actually review again.

**Comment by algonell** [comment]
My bad, one of my cleanups (occurence -> occurrence) was in aclocal.m4 (which is generated).

Thank for your patience and the detailed explanation!

**Comment by picnixz** [comment]
It's fine, took me a while as well in the beginning to see that I needed to regen configure with a specific version of autoconf...

**Comment by picnixz** [comment]
I'm not sure we want to fix ALL typos in one PR. Multiple small PRs would be more welcomed, both for helping reviews (don't open them yet until someone else confirmed this). I took the liberty of removing the code owners' review requests pending the decision.

File: Tools/scripts/combinerefs.py [hunk 2]
```diff
@@ -39,10 +39,10 @@
 repr is repr(object), extracted from the first PYTHONDUMPREFS output block.
 CAUTION:  If object is a container type, it may not actually contain all the
 objects shown in the repr:  the repr was captured from the first output block,
-and some of the containees may have been released since then.  For example,
+and some of the containers may have been released since then.  For example,
```
Comment (picnixz) [inline Tools/scripts/combinerefs.py]: `containee` is the right term, it's for something that is being contained, not something that is containing it.

File: Lib/test/test_weakref.py [hunk 3]
```diff
@@ -1860,7 +1860,7 @@ def test_weak_keyed_bad_delitem(self):
         self.assertRaises(KeyError, d.__delitem__, o)
         self.assertRaises(KeyError, d.__getitem__, o)
 
-        # If a key isn't of a weakly referencable type, __getitem__ and
+        # If a key isn't of a weakly referenceable type, __getitem__ and
```
Comment (picnixz) [inline Lib/test/test_weakref.py]: I think we already had this discussion somewhere, but referencable is accepted.

**Comment by algonell** [comment]
On my side, cleanups completed.

File: Lib/test/test_genexps.py [hunk 4]
```diff
@@ -265,7 +265,7 @@
     >>> me.gi_running
     0
 
-Verify that genexps are weakly referencable
+Verify that genexps are weakly referenceable
```
Comment (picnixz) [inline Lib/test/test_genexps.py]: ditto

File: Lib/test/test_itertools.py [hunk 5]
```diff
@@ -1857,7 +1857,7 @@ def _tee(iterator, link):
             t3 = tnew(t1)
             self.assertTrue(list(t1) == list(t2) == list(t3) == list('abc'))
 
-        # test that tee objects are weak referencable
+        # test that tee objects are weak referenceable
```
Comment (picnixz) [inline Lib/test/test_itertools.py]: ditto

File: Lib/test/test_generators.py [hunk 6]
```diff
@@ -2157,7 +2157,7 @@ def printsolution(self, x):
 """
 
 weakref_tests = """\
-Generators are weakly referencable:
+Generators are weakly referenceable:
```
Comment (picnixz) [inline Lib/test/test_generators.py]: ditto

File: Include/refcount.h [hunk 7]
```diff
@@ -401,7 +401,7 @@ static inline Py_ALWAYS_INLINE void Py_DECREF(PyObject *op)
  * There are cases where it's safe to use the naive code, but they're brittle.
  * For example, if `op` points to a Python integer, you know that destroying
  * one of those can't cause problems -- but in part that relies on that
- * Python integers aren't currently weakly referencable.  Best practice is
+ * Python integers aren't currently weakly referenceable.  Best practice is
```
Comment (picnixz) [inline Include/refcount.h]: ditto

**Review by vstinner — APPROVED** [review]
LGTM

**Review by vstinner — COMMENTED** [review]
Changes on Objects/mimalloc/ should be made upstream in the https://github.com/microsoft/mimalloc/ project.

**Review by picnixz — APPROVED** [review]
Out of curiosity, how do you look for spelling mistakes? do you use hunspell or aspell (or something else?)

**Comment by vstinner** [comment]
Usually, I dislike backporting such changes to stable branches. But since this change modifies many files, I prefer to backport it to reduce the risk of conflicts of future backports.

**Comment by vstinner** [comment]
Merged, thanks.

**Comment by algonell** [comment]
> Out of curiosity, how do you look for spelling mistakes? do you use hunspell or aspell (or something else?)

[codespell](https://github.com/codespell-project/codespell/)

**Comment by terryjreedy** [comment]
"you don't need to merge main into your branch unless you have conflicts."  One does if one wants to test a revised PR against current main on one's own machine before pushing to github.  However, one should 'git merge upstream/main' and not force-push.  And testing is not an issue for typos in comments and docstrings.

**Pull request merged**